lib/actions/reassemble.js

```diff
@@ -17,10 +18,14 @@ async function processAction(msg) {
     groupsSeen[groupId] = {
       groupSize,
       messageIdsSeen: new Set(),
+      incomingData: {},
     };
   }
 
   groupsSeen[groupId].messageIdsSeen.add(messageId);
+  if (messageData) {
+    groupsSeen[groupId].incomingData[messageId] = messageData;
+  }
```

Comment:
```suggestion
  groupsSeen[groupId].incomingData[messageId] = messageData;
```

I think we can skip the `if (messageData)` check and assign the `messageData` variable regardless. It could be that messageData is `false` at which point it may make sense to pass it through.

Reply:
was changed to `if (messageData !== undefined)`, otherwise incoming data includes an object with message id key with value = undefined. Or I should leave undefined value as well?

Reply:
You can leave undefined as well since when a JS object is converted to JSON, properties with `undefined` will be not included in the output JSON.

Reply:
done